hkmc2/shared/src/main/scala/hkmc2/codegen/HandlerLowering.scala

```diff
@@ -213,513 +175,413 @@ class HandlerLowering(paths: HandlerPaths, opt: EffectHandlers)(using TL, Raise,
-    
-    FunDefn(f.owner, f.sym, f.dSym, f.params, translateBlock(f.body,
-      f.params.flatMap(_.paramSyms).toSet,
-      callSelf,
-      L(f.sym),
-      functionHandlerCtx(s"Cont$$func$$${symToStr(f.sym)}$$", f.sym.nme))
-    )(forceTailRec = f.forceTailRec)
-  
-  private def translateBody(cls: ClsLikeBody, sym: BlockMemberSymbol)(using HandlerCtx): ClsLikeBody =
-    val curCtorCtx =
-      if handlerCtx.isTopLevel
-      then 
-        topLevelCtx(s"Cont$$modCtor$$${symToStr(sym)}$$", s"‹constructor of ${sym.nme}›")
-      else ctorCtx(
-        cls.isym.asPath,
-        s"Cont$$ctor$$${symToStr(sym)}$$", s"‹constructor of ${sym.nme}›")
-    ClsLikeBody(
-      cls.isym,
-      cls.methods.map(translateFun),
-      cls.privateFields,
-      cls.publicFields,
-      translateBlock(cls.ctor, Set.empty, N, R(cls.isym), curCtorCtx),
-    )
-  
-  private def translateCls(cls: ClsLikeDefn)(using HandlerCtx): ClsLikeDefn =
-    val curCtorCtx = ctorCtx(
-      cls.isym.asPath,
-      s"Cont$$ctor$$${symToStr(cls.sym)}$$", s"‹constructor of ${cls.sym.nme}›")
-    cls.copy(methods = cls.methods.map(translateFun),
-      ctor = translateBlock(cls.ctor, Set.empty, N, R(cls.isym), curCtorCtx),
-      companion = cls.companion.map(translateBody(_, cls.sym))) // TODO: callSelf
-  
-  // Handle block becomes a FunDefn and CallPlaceholder
-  private def translateHandleBlock(h: HandleBlock)(using HandlerCtx): Block =
-    val sym = BlockMemberSymbol(s"handleBlock$$", Nil)
-    val tSym = TermSymbol.fromFunBms(sym, N)
-    val lbl = freshTmp("handlerBody")
-    val lblLoop = freshTmp("handlerLoop")
-    
-    val handlerBody = translateBlock(
-      h.body, Set.empty, S(Call(Value.Ref(sym, S(tSym)), Nil)(true, false, false)), L(sym),
-      HandlerCtx(
-        false, true,
-        s"Cont$$handleBlock$$${symToStr(h.lhs)}$$", N, 
-        handlerCtx.debugInfo.copy(debugNme = s"‹handler body of ${h.lhs.nme}›"), 
-        state => blockBuilder
-          .assignFieldN(state.res.asPath.contTrace.last, nextIdent, Instantiate(true, state.cls, state.uid.asArg :: Nil))
-          .ret(PureCall(paths.handleBlockImplPath, state.res.asPath :: h.lhs.asPath :: Nil))
+        case fun: FunDefn =>
+          if !h.allowDefn then
+            raise(WarningReport(msg"Unexpected nested function: lambdas may not function correctly." -> fun.sym.toLoc :: Nil, source = Diagnostic.Source.Compilation))
+          val (debugInfoSym, debugInfo, fun2) = translateFunLike(fun, Value.Ref(fun.sym, S(fun.dSym)), N, fun.sym.nme)
+          if opt.debug then Assign(debugInfoSym, Tuple(false, debugInfo), k(fun2)) else k(fun2)
+        case ClsLikeDefn(owner, isym, sym, ctorSym, kind, paramsOpt, auxParams, parentPath, methods, privateFields, publicFields, preCtor, ctor, companion, bufferable) =>
+          if !h.allowDefn then
+            raise(WarningReport(msg"Unexpected nested class: lambdas may not function correctly." -> isym.toLoc :: Nil, source = Diagnostic.Source.Compilation))
```

Comment:
Same as above. I realize some tests contain nested modules which aren't lifted, but those should probably be left as `:fixme`s.